### PR TITLE
Refactor NodesInfoRequest to extract NodeInfoMetrics to a top-level class from it

### PR DESCRIPTION
**Pull request by NEUpanning**
This commit extracts `NodeInfoMetrics` to a top-level class from `NodesInfoRequest`. `NodeInfoMetrics` is a container that encapsulates the necessary information needed to indicate which node information is requested.

Relates #99938

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/TransportNodesInfoAction.java [hunk 1]
```diff
@@ -76,41 +78,72 @@ protected NodeInfo newNodeResponse(StreamInput in, DiscoveryNode node) throws IO
...
+            metrics.contains(NodesInfoMetrics.Metric.HTTP.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.REMOTE_CLUSTER_SERVER.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.PLUGINS.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.INGEST.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.AGGREGATIONS.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.INDICES.metricName())
         );
     }
 
     public static class NodeInfoRequest extends TransportRequest {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/TransportNodesInfoAction.java:98]: The changes to `NodeInfoRequest` seem unrelated to the rest of this PR, would you revert them?

Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/TransportNodesInfoAction.java:98]: Thanks. The changes to `NodeInfoRequest` is to decouple it from `TransportNodesInfoAction.NodeInfoRequest`. Then we can trim down the metrics that `NodeInfoRequest` carries. Do you mean we just extract `NodeInfoMetrics` to a top-level class from NodesInfoRequest and then do that changes in the next PR?

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/TransportNodesInfoAction.java:98]: Right, let's _only_ extract the metrics to a top-level class first, then we can work on trimming down `NodeInfoRequest` in a follow-up.

Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/TransportNodesInfoAction.java]: done

**Review by DaveCTurner — APPROVED** [review]
Thanks, LGTM

**Comment by DaveCTurner** [comment]
@elasticmachine test this please
